### ISSUE TITLE
After 7.20.0 unable to set config in alpine / bullseye docker image

### ISSUE THREAD

**Issue by TomABrook** [pnpm/pnpm#5848]
Since the release of 7.20.0 our docker images are failing to build with the following error: 
```
#8 0.955  ERROR  Unable to find the global bin directory
#8 0.955 Run "pnpm setup" to create it automatically, or set the global-bin-dir setting, or the PNPM_HOME env variable. The global bin directory should be in the PATH.
```
Adding `pnpm setup` as suggested gives an unknown shell type error but I was able to work around with the suggestion [here](https://github.com/pnpm/pnpm/issues/4495#issuecomment-1297010385): 
```
RUN npm install --global pnpm \
    && SHELL=bash pnpm setup \
    && source /root/.bashrc
```
However with this I am still seeing the same "Unable to find the global bin directory" error once the script tries to set a config parameter.

I'm not sure if this is a bug or the result of an intentional change, but any suggestions on how get this working again would be much appreciated

### Code to reproduce the issue:

```
FROM node:16-bullseye-slim

RUN npm install -g npm@8
RUN npm install -g node-gyp
RUN npm install --global pnpm
RUN pnpm config set loglevel=error -g

ENTRYPOINT ["tail", "-f", "/dev/null"]
```

### Additional information:

 Tested with following base images:
- node:16-bullseye-slim
- node:14-alpine

**Comment by zkochan** [comment]
Fixed in v7.21

**Comment by TomABrook** [comment]
Great, thanks for the quick fix!
